OMCompiler/Compiler/NFFrontEnd/NFDimension.mo

```diff
@@ -597,14 +602,9 @@ public
   function simplify
     input output Dimension dim;
   algorithm
-    () := match dim
-      case EXP()
-        algorithm
-          dim.exp := SimplifyExp.simplify(dim.exp);
-        then
-          ();
-
-      else ();
+    dim := match dim
+      case EXP() then fromExp(dim.exp, Expression.variability(dim.exp));
```

Comment:
You lost the call to `SimplifyExp.simplify` here, so this isn't really doing anything now.